mirrord-layer/src/lib.rs

```diff
@@ -88,12 +88,43 @@ pub(crate) static ENABLED_UDP_OUTGOING: OnceLock<bool> = OnceLock::new();
 
 pub(crate) const MIRRORD_SKIP_LOAD: &str = "MIRRORD_SKIP_LOAD";
 
+/// Check if we're running in NixOS or Devbox
+/// if so, add `sh` to the skip list because of https://github.com/metalbear-co/mirrord/issues/531
+fn nix_devbox_patch(config: &mut LayerConfig) {
+    let mut current_skip = config
+        .skip_processes
+        .clone()
+        .map(VecOrSingle::to_vec)
+        .unwrap_or_default();
+    // already contains, doesn't matter
+    if current_skip.contains(&"sh".to_string()) {
+        return;
+    }
+
+    if is_nix_or_devbox() {
+        current_skip.push("sh".into());
+        std::env::set_var("MIRRORD_SKIP_PROCESSES", current_skip.join(";"));
+        config.skip_processes = Some(VecOrSingle::Multiple(current_skip));
+    }
+}
+/// Check if NixOS or Devbox by discrimnating env vars.
+fn is_nix_or_devbox() -> bool {
+    if let Ok(res) = std::env::var("IN_NIX_SHELL") && res.as_str() == "1" {
```

Comment:
nit: can't this be `if / else if / else` to drop the `return` keyword?

Reply:
ok